### PR TITLE
Deplete

### PR DESCRIPTION
**Pull request by Bar2005**
wrote the code we talked about to help the driver deplete

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java [hunk 1]
```diff
@@ -148,7 +157,12 @@ private static void setSubsystemToState(Gamepad gamepad1, Gamepad gamepad2) {
 
         lastYButtonState = gamepad1.y;
         lastRightBumperButtonState = gamepad1.right_bumper;
+        lastLeftBumperButtonState = gamepad1.left_bumper;
+        lastxButtonState = gamepad1.x;
         lastState = state;
+        if (!gamepad1.y){
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:163]: Why y is here instead of right bumper?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/elevator/Elevator.java [hunk 2]
```diff
@@ -48,6 +48,8 @@ public static void operate(ElevatorStates elevatorState, Gamepad gamepad1) {
             case OVERRIDE:
                 elevatorPower = gamepad1.right_stick_y;
                 break;
+            case DEPLETE:
+                elevatorPower = -0.1f;
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/elevator/Elevator.java:52]: Create a constant for this power.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java [hunk 3]
```diff
@@ -29,14 +29,17 @@ public class SubSystemManager {
...
     private static boolean yButtonControl = false; // TODO maybe a better name?
     private static boolean rightStickControl = false; // TODO maybe better name?
+    private static boolean lastLeftBumperButtonState;
+    private static boolean depleteControl = true;
 
     private static RobotState getState(Gamepad gamepad) {
         return gamepad.b ? RobotState.TRAVEL
                 : gamepad.a ? RobotState.INTAKE
-                        : gamepad.x ? RobotState.DEPLETE : gamepad.left_bumper ? RobotState.CLAWINTAKE : null;
+                        : gamepad.right_bumper ? RobotState.DEPLETE : gamepad.dpad_up ? RobotState.CLAWINTAKE : null;
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:42]: Why right bumper?

**Review by Ido-Sobol — CHANGES_REQUESTED** [review]
Pls write down what each button that you changed in this pr supposed to do, it will make the reviewing process easier for me 😀

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java (hunk 3 above)
Comment (Bar2005) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:42]: thats what we talked about. Whenever shemi prreses the right bumber button the elavator returns to the position from the second driver. We chose this button because he is relatively close to the triangle button

**Comment by Ido-Sobol** [comment]
@Bar2005 did you checked it?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java [hunk 4]
```diff
@@ -23,7 +23,7 @@
 @TeleOp(name = "main")
 public class Robot extends LinearOpMode {
 
-    ElapsedTime time = new ElapsedTime();
+    ElapsedTime robotTime = new ElapsedTime();
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java:26]: Why did you change this variable's name?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java [hunk 5]
```diff
@@ -53,15 +59,20 @@ public static ElevatorStates getElevatorStateFromSecondDriver(Gamepad gamepad2)
                         : gamepad2.x ? ElevatorStates.MID : gamepad2.y ? ElevatorStates.HIGH : null;
     }
 
+
     private static void setSubsystemToState(Gamepad gamepad1, Gamepad gamepad2) {
+        float currentTime = (float) subSystemManagerTime.milliseconds();
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:64]: If you are using the currentTime in more then 1 place(robot,subsystem manager), you should move it to globalData, and then update it in robot.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java [hunk 6]
```diff
@@ -78,7 +89,7 @@ private static void setSubsystemToState(Gamepad gamepad1, Gamepad gamepad2) {
                 }
                 break;
             case INTAKE:
-                if (!GlobalData.hasGamePiece) {
+                if (GlobalData.hasGamePiece) {
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:92]: Is hasGamePiece updated somewhere?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/claw/Claw.java [hunk 7]
```diff
@@ -6,7 +6,7 @@
 public class Claw {
 
     private static float pos;
-    public static final Servo clawServo;
+    public static Servo clawServo;
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/claw/Claw.java:9]: Why remove the final?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/elevator/Elevator.java [hunk 8]
```diff
@@ -48,6 +48,8 @@ public static void operate(ElevatorStates elevatorState, Gamepad gamepad1) {
             case OVERRIDE:
                 elevatorPower = gamepad1.right_stick_y;
                 break;
+            case DEPLETE:
+                elevatorPower = ElevatorConstants.depletePower;
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/elevator/Elevator.java:52]: I don't think deplete should be it's own elevator state, I think it should be part of override, and then when we are depleting we are going to override and adding -0.1 to what the driver is asking.

**Review by Ido-Sobol — CHANGES_REQUESTED** [review]

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java (hunk 4 above)
Comment (Bar2005) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java:26]: it is a more acuurate name, dont you think so?

Comment (Ido-Sobol) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java:26]: No.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java (hunk 5 above)
Comment (Bar2005) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:64]: actually there is no nedd of this, I will remove anything associated with this

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/claw/Claw.java (hunk 7 above)
Comment (Bar2005) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/claw/Claw.java:9]: because once there is a final there you cant declare the hardware of this servo which leads to an error
ok?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/elevator/Elevator.java (hunk 8 above)
Comment (Bar2005) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/elevator/Elevator.java:52]: there will be a problem since you will be have to add another condition and in it to call elevator.operate once again
do you prefer this option?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/claw/Claw.java (hunk 7 above)
Comment (Ido-Sobol) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/claw/Claw.java:9]: Sure

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/elevator/Elevator.java (hunk 8 above)
Comment (Ido-Sobol) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/elevator/Elevator.java:52]: Robot state should be global, then you can check if we are depleting.
